### PR TITLE
CORE-610 Autoscaling implemented in web app chart

### PR DESCRIPTION
**Pull request by kobrikx**
#### What's new:

- HPA added

#### Testing done:

Screenshot 2023-12-14 at 13 08 43
Screenshot 2023-12-14 at 13 11 09
Screenshot 2023-12-14 at 13 12 45
Screenshot 2023-12-14 at 13 13 37

### CODE REVIEW

File: charts/web/templates/deployment.yaml [hunk 1]
```diff
@@ -9,7 +9,9 @@ metadata:
     link.argocd.argoproj.io/external-link: https://{{ .Values.global.app.name}}.{{ .Values.global.env}}.{{.Values.global.app.domain}}
 spec:
   # Replicas count
```
Comment (DimmKirr) [inline charts/web/templates/deployment.yaml]: Seems like this comment should be within the conditional (after the next line), no?

File: charts/web/values.yaml [hunk 2]
```diff
@@ -28,10 +28,34 @@ global:
...
+## Number of pod replicas(Disabled when HPA enabled):
     replicas: 1
 
+## Horizontal pod autoscaling settings:
+  hpa:
+    enabled: false
+    minReplicas: 2
+    maxReplicas: 4
+    behavior:
+      scaleUp:
```
Comment (DimmKirr) [inline charts/web/values.yaml:43]: Please add some description what's our intended logic for the autoscaler. When would it scale up, when would it scale down? Just more information on our intentions please.

File: charts/web/templates/deployment.yaml [hunk 3]
```diff
@@ -9,7 +9,9 @@ metadata:
...
+  {{- if not .Values.global.hpa.enabled }}
   replicas: {{ .Values.global.app.replicas}}
```
Comment (DimmKirr) [inline charts/web/templates/deployment.yaml:13]: How would it work when hpa is off? Would replicas be undefined? Is that intended?

**Review by DimmKirr — CHANGES_REQUESTED** [review]
Thanks, I have couple questions, please address.

File: charts/web/templates/deployment.yaml (hunk 3 above)
Comment (kobrikx) [reply charts/web/templates/deployment.yaml:13]: @AutomationD When we use HPA - we don't need the Replicas parameter. Therefore, it appears in the deployment chart only when HPA is disabled.

File: charts/web/values.yaml [hunk 4]
```diff
@@ -28,10 +28,40 @@ global:
     region: aws-region
     containerName: ""
     domain: domain.local
-
-## Number of pod replicas:
+## Number of pod replicas(Disabled when HPA enabled):
```
Comment (DimmKirr) [inline charts/web/values.yaml]:
```suggestion
## Number of pod replicas(Disabled when HPA is enabled):
```

**Review by DimmKirr — APPROVED** [review]
Thank you, looks good to me. please address minor typos and feel free to merge without an extra review.

**Pull request merged**